### ISSUE TITLE
Overloaded function signatures don't fully qualify types

### ISSUE THREAD

**Issue by dirn** [python/mypy#7095]
This is a request for a future that will provide more useful output when calling overloaded functions.

When I check the following code with mypy

```
from a import MyType as MyTypeA
from b import MyType as MyTypeB

def f(a: MyTypeA) -> MyTypeB:
    return a
```

I get

```
error: Incompatible return value type (got "a.MyType", expected "b.MyType")
```

The message includes the fully qualified name for the types that don't match. When I check this code with mypy, though

```
from typing import overload, Union

from a import MyType as MyTypeA
from b import MyType as MyTypeB

@overload
def f(a: MyTypeA) -> MyTypeB:
    pass

@overload
def f(a: MyTypeB) -> MyTypeB:
    pass

def f(a: Union[MyTypeA, MyTypeB]) -> MyTypeB:
    return a

def g():
    f(MyTypeA())
```

I get

```
error: Incompatible return value type (got "Union[MyType, MyType]", expected "MyType")
```

This message only includes the names of the types, not the fully qualified name. (This also applies to the `No overload variant` message that is sometimes output.) The information is really useful for understanding why the type mismatch occurs.

The above output was generated using mypy 0.711 and CPython 3.7.2, but I can also reproduce the error using 46aa00d8a8d5c870c6ef24b82261b6b257e2ea03 against code written for CPython 2.7.x.

**Comment by ilevkivskyi** [comment]
(Setting priority to high since the error is really obscure, and is probably not hard to fix.)

**Comment by nkashy1** [comment]
@ilevkivskyi : I made a PR which adds a test for this issue: https://github.com/python/mypy/pull/7490

That's as far as I got, because the test passes.

Not sure if you want to merge in that test, but it's there if you want it.

**Comment by msullivan** [comment]
This was fixed by #7226. Thanks for triaging this, @nkashy1!
